actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -227,16 +228,22 @@ def digest_path_from_template(template) # :nodoc:
       end
 
     private
-      def fragment_name_with_digest(name, virtual_path, digest_path)
+      def fragment_name_with_digest(name, virtual_path, digest_path, container: nil)
         virtual_path ||= @virtual_path
 
         if virtual_path || digest_path
           name = controller.url_for(name).split("://").last if name.is_a?(Hash)
 
           digest_path ||= digest_path_from_template(@current_template)
 
-          [ digest_path, name ]
+          if container
+            container << digest_path
+            container << name
+          else
+            [ digest_path, name ]
+          end
         else
+          container << name if container
           name
         end
```

Comment:
Don't think this is worth saving the extra array allocation. Curious if we could replace all this with something like:

```ruby
view.cache_fragment_name(item, virtual_path: template.virtual_path, digest_path: digest_path).tap do |key_parts|
  key.concat Array(key_parts)
end
```

in the collection caching stuff. Fine if you want to replace `Array()` with more granular handling.